pkg/model/iam/iam_builder.go

```diff
@@ -338,19 +338,27 @@ func (b *PolicyBuilder) AddS3Permissions(p *Policy) (*Policy, error) {
 						),
 					})
 				} else if b.Role == kops.InstanceGroupRoleNode {
+					resources := []string{
+						strings.Join([]string{b.IAMPrefix(), ":s3:::", iamS3Path, "/addons/*"}, ""),
+						strings.Join([]string{b.IAMPrefix(), ":s3:::", iamS3Path, "/cluster.spec"}, ""),
+						strings.Join([]string{b.IAMPrefix(), ":s3:::", iamS3Path, "/config"}, ""),
+						strings.Join([]string{b.IAMPrefix(), ":s3:::", iamS3Path, "/instancegroup/*"}, ""),
+						strings.Join([]string{b.IAMPrefix(), ":s3:::", iamS3Path, "/pki/issued/*"}, ""),
+						strings.Join([]string{b.IAMPrefix(), ":s3:::", iamS3Path, "/pki/private/kube-proxy/*"}, ""),
+						strings.Join([]string{b.IAMPrefix(), ":s3:::", iamS3Path, "/pki/ssh/*"}, ""),
+						strings.Join([]string{b.IAMPrefix(), ":s3:::", iamS3Path, "/secrets/dockerconfig"}, ""),
+					}
+
+					if !b.UseBootstrapTokens() {
+						resources = append(resources, strings.Join([]string{b.IAMPrefix(), ":s3:::", iamS3Path, "/pki/private/kubelet/*"}, ""))
```

Comment:
You could maybe simplify the logic below where there's the addition of access to `/pki/private/node-authorizer-client`, when `UseBootstrapTokens()` is true. Otherwise LGTM :)

Reply:
hi @justinsb .. so this was added for existing clusters whom are migrating to bootstrap tokens .. Since we weren't deleting the keypair I wanted to ensure the nodes weren't able to gain access to the certificate anymore .. Alternatively we'd have to inform the users to delete the certificate post upgrading ..

Reply:
Good point - I rewrote it to use the same logic also for bootstrap tokens.

For users not adopting bootstrap tokens, I wanted to avoid the spurious IAM policy change where we broke out the kubelet path into its own rule.  But with the reorg that @KashifSaadat suggested I think it's now clearer what's going on here...